zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/message/MessagePublishProcessor.java

```diff
@@ -104,7 +111,14 @@ public void processRecord(final TypedRecord<MessageRecord> command) {
     }
 
     messageRecord = command.getValue();
-
+    if (routingInfo.partitionForCorrelationKey(messageRecord.getCorrelationKeyBuffer())
+        != partitionId) {
+      final var reason =
+          "The message has not been routed to the right partition. This is probably a temporary issue, please retry in a few seconds";
+      rejectionWriter.appendRejection(command, RejectionType.INVALID_STATE, reason);
```

Comment:
:thinking: Would it be possible to translate to a specific error code that helps users to implement automatic retry?

Reply:
This would a potentially breaking change if the client code is not handling unknown values from the enum. 
I would say that it should be done in a follow up PR :thinking: